logback/src/main/java/com/grafana/loki/HttpHandler.java

```diff
@@ -0,0 +1,60 @@
+package com.grafana.loki;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.Builder;
+import java.net.http.HttpResponse;
+
+public class HttpHandler {
+	private static HttpClient client;
+
+	public static void sendLoki(String tenantID, URI url, byte[] body) {
+		int retryCount = 3;
+		int retryDelay = 2000;
+		if (client == null) {
+			client = HttpClient.newHttpClient();
+		}
+
+		Builder builder = HttpRequest.newBuilder().uri(URI.create(url.toString()))
+				.POST(HttpRequest.BodyPublishers.ofByteArray(body));
+		builder.header("Content-Type", "application/x-protobuf");
+
+		if (tenantID != "") {
+			builder.header("X-Scope-OrgID", tenantID);
+		}
+		HttpRequest request = builder.build();
+
+		RetryOnException retryHandler = new RetryOnException(retryCount, retryDelay);
+		while (true) {
```

Comment:
I think for certain errors like 500's this will currently retry forever

Reply:
I think maybe  `while(retryHandler.shouldRetry())` would work here

Reply:
Yep, will make the change